src/braket/pennylane_plugin/ahs_device.py

```diff
@@ -440,15 +439,202 @@ class BraketLocalAquilaDevice(BraketAhsDevice):
+                max_index = int(np.argmax(time_slice))
+                max_detuning = detunings[max_index]
+
+                pattern = [det / _max for det in time_slice]
+                break
+
+        # If all time steps are zero, local detuning is assumed to be zero
+        if pattern == []:
+            max_detuning = 0
+            pattern = [1.0] * len(detunings)
+
+        # Validate that detunings follow pattern along all time steps
+        for i, t in enumerate(time_points):
+            time_slice = evaluated_detunings[:, i]
+            
+            new_time_slice = [p * float(max_detuning(t * 1e6)) for p in pattern]
+            if not np.allclose(time_slice, new_time_slice):
+                raise ValueError(
+                    "Local detunings don't match. Make sure that all local detunings match "
+                    "in shape and only differ in magnitude."
+                )
+
+        return max_detuning, Pattern(pattern)
+
+    def _convert_pulses_to_shifting_field(self, detuning, pattern, time_interval):
+        """Uses the list of ``RydbergPulse`` objects from PennyLane to create a ``ShiftingField`` object from
+        AWS Braket.
+
+        Args:
+            detuning (callable): detuning for the local drives
+            pattern (Pattern): list containing magnitude of detuning for all atoms in the device
+            time_interval(array[Number, Number]]): The start and end time for the applied pulses
+
+        Returns:
+            ShiftingField: the object representing the local drive for the AnalogueHamiltonianSimulation object
+        """
+        time_points = self._get_sample_times(time_interval)
+        ts_detuning = self._convert_to_time_series(
+            detuning, time_points, scaling_factor=2 * np.pi * 1e6
+        )
+        shift = ShiftingField(magnitude=Field(time_series=ts_detuning, pattern=pattern))
+
+        return shift
+
+    def _validate_pulses(self, pulses):
+        """Validate that all pulses are defined as expected by the device. This validation includes:
+
+        * Verifying that a global drive is present
+        * Verifying that all local pulses have zero amplitude and phase
+        * Verifying that there are no overlapping wires among the local drives
+        * Verifying that all local detunings are of the same type (float or callable)
+
+        Args:
+            pulses (List[RydbergPulse]): List containing all pulses
+
+        Raises:
+            ValueError: if pulses are invalid
+        """
+
+        # Iterate through pulses to find global drive
+        global_index = -1
+        for i, pulse in enumerate(pulses):
+            if set(pulse.wires) == set(self.wires):
+                if global_index != -1:
+                    raise ValueError(
+                        "Cannot execute a ParametrizedEvolution with multiple global drives."
+                    )
+                global_index = i
+            elif not set(pulse.wires).issubset(set(self.wires)):
```

Comment:
```suggestion
            elif not self.wires.contains_wires(pulse.wires):
```
probably does the same internally